Exercise_1.java

```diff
@@ -1,32 +1,46 @@
+package com.s30;
 class Stack { 
     static final int MAX = 1000; 
     int top; 
     int a[] = new int[MAX]; // Maximum size of Stack 
   
     boolean isEmpty() 
     { 
-        //Write your code here 
+    	 return top == -1;
     } 
 
     Stack() 
     { 
+    	a = new int[MAX];
+
+    	top = -1;
         //Initialize your constructor 
     } 
   
     boolean push(int x) 
     { 
+    	a[++top] = x;
+    	if(isFull())
+    		return false;

```

Comment:
The StackOverflow should be checked before adding. This way even on a full stack the program will add the value and then return False.